src/app/core/services/signing/signing.service.ts

```diff
@@ -57,6 +58,19 @@ export class SigningService {
       const derSignature = signature.toDER();
       const signatureBase64 = naclUtil.encodeBase64(derSignature);
 
+      return signatureBase64;
+    } else if (type == SignatureType.RSA) {
+      const signAlgorithm = {
+        name: "RSASSA-PKCS1-v1_5",
+        modulusLength: 4096,
+        publicExponent: new Uint8Array([0x01, 0x00, 0x01]),
+        hash: { name: "SHA-256" }
+      };
+
+      const privateCryptoKey = await window.crypto.subtle.importKey("pkcs8", convertPemToBinary(privateKey) as ArrayBuffer, signAlgorithm, true, ["sign"]);
+      const signature = await window.crypto.subtle.sign(signAlgorithm, privateCryptoKey, dataToSign);
+      const signatureBase64 = arrayBufferToBase64String(signature);
```

Comment:
In the future, it might be worth it to support multiple encodings for the signature with some flag passed to this function. Just something to keep in mind.